### PR TITLE
Fix: Static assets

### PR DESCRIPTION
**Pull request by Deokumarjnu**
# Description

Added support for compiling custom/static assets.

Fixes # https://github.com/quintype/ace-planning/issues/17

Dependencies # NA

Documentation # (link to the corresponding documentation changes)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Tested it locally on prothomalo
- [x] Tested it on prothomalo staging

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Jeevan-Kishore — APPROVED** [review]

File: config/quintype-build.js [hunk 1]
```diff
@@ -1,11 +1,22 @@
 const { produce } = require("immer");
 const { BundleAnalyzerPlugin } = require("webpack-bundle-analyzer");
+const fs = require("fs");
 
 module.exports = {
   modifyWebpackConfig: function({ defaultConfig }) {
     const config = produce(defaultConfig, function(draft) {
       draft.node = { Buffer: false };
       draft.entry["font"] = "./app/client/font.js";
+      const dir = "./app/assets/fonts";
```
Comment (nandap4790) [inline config/quintype-build.js]: @Deokumarjnu @cricket-lover are we just handling for fonts ?

**Review by nandap4790 — CHANGES_REQUESTED** [review]

File: config/quintype-build.js [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
+      try {
+        fs.readdirSync(dir).forEach(file => {
+          const fullPath = `${dir}/${file}`;
+          draft.entry[file] = fullPath;
+        });
+      } catch (error) {
+        console.log("Error !!!", error);
```
Comment (ags1773) [inline config/quintype-build.js]: Why is try/catch needed here?

Comment (ags1773) [inline config/quintype-build.js:17]: Why is try/catch needed here?

Comment (Deokumarjnu) [reply config/quintype-build.js]: No need here, but added for sefty, by any chance if some error occur while reading the files , it should able to handle it.

File: config/quintype-build.js [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
+        const files = await Promise.all(
+          dirents.map(dirent => {
+            const res = resolve(dir, dirent.name);
+            return dirent.isDirectory()
+              ? getFiles(res)
+              : { fileName: dirent.name, filePath: res };
+          })
+        );
+        return files.flat();
+      }
```
Comment (Deokumarjnu) [inline config/quintype-build.js:23]: Recursively going through all the files and folders within the given directory.

File: config/quintype-build.js [hunk 4]
```diff
@@ -1,11 +1,33 @@
...
+
+      const dir = "./app/static-assets";
+
+      getFiles(dir).then(files =>
+        files.forEach(file => (draft.entry[file.fileName] = file.filePath))
+      );
```
Comment (Deokumarjnu) [inline config/quintype-build.js:29]: making entry point for all the files within static-assets folder.

File: config/quintype-build.js [hunk 5]
```diff
@@ -1,11 +1,31 @@
...
+          return dirent.isDirectory()
+            ? getFiles(res)
+            : { fileName: dirent.name, filePath: res };
+        });
+        return files.flat();
+      }
+      const files = getFiles(dir);
+      files.forEach(file => {
+        draft.entry[file.fileName] = file.filePath;
+      });
```
Comment (Deokumarjnu) [inline config/quintype-build.js]: making entry point for all the files within static-assets folder.

File: config/quintype-build.js [hunk 6]
```diff
@@ -1,11 +1,31 @@
...
+      function getFiles(dir) {
+        const dirents = fs.readdirSync(dir, { withFileTypes: true });
+        const files = dirents.map(dirent => {
+          const res = resolve(dir, dirent.name);
+          return dirent.isDirectory()
+            ? getFiles(res)
+            : { fileName: dirent.name, filePath: res };
+        });
+        return files.flat();
+      }
```
Comment (Deokumarjnu) [inline config/quintype-build.js:23]: Recursively going through all the files and folders within the given directory.

**Pull request closed without merging**